src/arviz_plots/plots/lm_plot.py

```diff
@@ -0,0 +1,324 @@
+from collections.abc import Mapping, Sequence
+from copy import copy
+from importlib import import_module
+from typing import Any, Literal
+
+import numpy as np
+import xarray as xr
+from arviz_base import rcParams
+from scipy.interpolate import griddata
+from scipy.signal import savgol_filter
+
+from arviz_plots.plot_collection import PlotCollection
+from arviz_plots.plots.utils import (
+    filter_aes,
+    get_contrast_colors,
+    process_group_variables_coords,
+    set_wrap_layout,
+)
+from arviz_plots.visuals import fill_between_y, labelled_x, labelled_y, line_xy, scatter_xy
+
+
+def plot_lm(
+    dt,
+    target_data=None,
+    independent_data=None,
+    prediction_data=None,
+    ci_mean_independent_data=None,
+    filter_vars=None,
+    prediction_group="predictions",
+    coords=None,
+    sample_dims=None,
+    ci_prob=None,
+    ci_kind=None,
+    smooth=True,
+    plot_collection=None,
+    backend=None,
+    aes_by_visuals: Mapping[
+        Literal[
+            "line",
+            "mean_line",
+            "fill",
+            "scatter",
+            "xlabel",
+            "ylabel",
+        ],
+        Sequence[str],
+    ] = None,
+    visuals: Mapping[
+        Literal[
+            "line",
+            "mean_line",
+            "fill",
+            "scatter",
+            "xlabel",
+            "ylabel",
+        ],
+        Mapping[str, Any] | Literal[False],
+    ] = None,
+    stats: Mapping[
+        Literal["credible_interval",],
+        Mapping[str, Any] | xr.Dataset,
+    ] = None,
+    **pc_kwargs,
+):
+    """Posterior predictive and mean plots for regression-like data..
+
+    Parameters
+    ----------
+    dt : DataTree
+        Input data
+
+    """
+    if sample_dims is None:
+        sample_dims = rcParams["data.sample_dims"]
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+    if visuals is None:
+        visuals = {}
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    if backend is None:
+        if plot_collection is None:
+            backend = rcParams["plot.backend"]
+        else:
+            backend = plot_collection.backend
+    if ci_prob is None:
+        ci_prob = rcParams["stats.ci_prob"]
+    if ci_kind is None:
+        ci_kind = rcParams["stats.ci_kind"] if "stats.ci_kind" in rcParams else "eti"
+
+    if aes_by_visuals is None:
+        aes_by_visuals = {}
+    else:
+        aes_by_visuals = aes_by_visuals.copy()
+
+    if stats is None:
+        stats = {}
+    else:
+        stats = stats.copy()
+
+    if target_data is None:
+        target_data = process_group_variables_coords(
+            dt, group="observed_data", var_names="y", filter_vars=filter_vars, coords=coords
+        )
+    elif isinstance(target_data, str):
+        target_data = process_group_variables_coords(
+            dt, group="observed_data", var_names=target_data, filter_vars=filter_vars, coords=coords
+        )
```

Comment:
as written now, this can be reduce to the elif as an if instead and `target_data="y"` in the call signature. I think we can probably have a better default if we discuss the behaviour tomorrow (and potentially support multiple variables) so I would change it to something like:

```
if target_data is None:
    target_data = "y"
if not isinstance(target_data, xr.DataArray | xr.Dataset):
    target_data = process_group...
```